frontend/src/app/services/api.service.ts

```diff
@@ -108,42 +109,32 @@ export class ApiService {
     }
 
     // Tasks
-    public getTasksStatus<ExpectedResult>(
+    public getTasksStatus(
         tasks: TaskResult
-    ): Promise<TasksOutcome<ExpectedResult>> {
+    ): Observable<TasksOutcome> {
         return this.http
-            .post<TasksOutcome<ExpectedResult>>('/api/task_status', tasks)
-            .toPromise();
+            .post<TasksOutcome>('/api/task_status', tasks);
     }
 
     public abortTasks(data: TaskResult): Promise<TaskSuccess> {
         return this.http
-            .post<TaskSuccess>('/api/task_status', data)
+            .post<TaskSuccess>('/api/abort_tasks', data)
             .toPromise();
     }
 
-    private tasksDone<ExpectedResult>(response: TasksOutcome<ExpectedResult>) {
-        return response.status !== 'working';
+    private tasksDone(response: TasksOutcome) {
+        return response.status === 'done';
```

Comment:
how will a failed task stop polling?

Reply:
It throws a hard error now from the backend.